test/libsolidity/lsp/goto_definition.sol

```diff
@@ -29,40 +33,211 @@ contract C
     function virtual_inheritance() public payable
     {
         obj = new IA();
+        //        ^ @usingIASymbol
         obj.f(1); // goto-definition should jump to definition of interface.
+    //      ^ @virtualFunctionLookup
     }
 
     using IntLib for *;
     function using_for(int i) pure public
     {
         i.add(5);
+ //       ^ @usingIntAdd
         14.add(4);
     }
 
     function useLib(uint n) public payable returns (uint)
     {
         return Lib.add(n, 1);
+        //     ^ @LibSymbol
+        //         ^ @LibAddSymbol
     }
 
     function enums(Color c) public pure returns (Color d)
+    //             ^ @ColorSymbolInParameter
     {
         Color e = Color.Red;
+        //    ^ @eVariableDeclaration
+        //              ^ @RedEnumMemberAccess
         if (c == e)
+        //       ^ @eVariableAccess
             d = Color.Green;
         else
             d = c;
     }
 
     type Price is uint128;
+    //   ^^^^^ @PriceDeclaration
     function udlTest() public pure returns (uint128)
     {
         Price p = Price.wrap(128);
+    //  ^ @PriceSymbol
+    //            ^ @PriceInWrap
         return Price.unwrap(p);
     }
 
     function structCtorTest(uint8 v) public pure returns (uint8 result)
     {
         RGBColor memory c = RGBColor(v, 2 * v, 3 * v);
+        //                       ^ @RGBColorCursor
         result = c.red;
+        int a;
+//      ^^^^^ @unusedLocalVar
     }
 }
+// ----
+// goto_definition: @unusedLocalVar 2072
+// lib: @diagnostics 2072
+// -> textDocument/definition {
+//     "position": @importDirective
+// }
+// <- {
+//     "jsonrpc": "2.0",
+//     "result": [
+//         {
+//             "range": {
+//                 "end": {
+//                     "character": 0,
+//                     "line": 0
+//                 },
+//                 "start": {
+//                     "character": 0,
+//                     "line": 0
+//                 }
+//             },
+//             "uri": "lib.sol"
+//         }
+//     ]
+// }
+// -> textDocument/definition {
+//     "position": @usingIASymbol
+// }
+// <- {
+//     "jsonrpc": "2.0",
+//     "result": [
```

Comment:
Forward message is slightly abstracted. Good.
Backward message is however bare, which is probably a little bit confusing. I'd suggest to only require passing the response value, such as I'm going to suggest below.